### PR TITLE
Feature/account states framwork

### PR DESCRIPTION
**Pull request by schalky**
Created framework for events that can be triggered at startup, home screen (unused at the moment) and pre-circle.
Currently 2 examples of events, startup has a profile check event and pre-circle is the onboarding check.

### CODE REVIEW

File: lib/app/circle/circle_session_page.dart [hunk 1]
```diff
@@ -167,16 +167,12 @@ class CircleSessionPageState extends ConsumerState<CircleSessionPage>
...
-        final userState = await repo.userAccountState();
-        if (!userState.boolAttribute(AccountState.onboarded)) {
-          if (!mounted) return;
-          await OnboardingScreen.showOnboarding(context,
-              onComplete: (bool result) {
-            // show
-            Navigator.of(context).pop();
-          });
-        }
+        final accountState = await repo.userAccountState();
```
Comment (ansells) [inline lib/app/circle/circle_session_page.dart:172]: Is this still a thing?

**Review by ansells — APPROVED** [review]

**Comment by blopker** [comment]
Looks good! I couldn't tell, but will this activate the onboarding modal when merged, or is that initialization still commented out?

File: lib/app/circle/circle_session_page.dart (hunk 1 above)
Comment (schalky) [reply lib/app/circle/circle_session_page.dart:172]: that code was disabled per request. Can be re-enabled if we want onboarding to show.

**Review by blopker — APPROVED** [review]
Looks great!

**Pull request merged**